File: test-db.js
require('dotenv').config();
const { Client } = require('pg');

async function test() {
  const dbUser = process.env.DB_USER || 'postgres';
  const dbPassword = process.env.DB_PASSWORD;
  const dbHost = process.env.DB_HOST || 'localhost';
  const dbPort = process.env.DB_PORT || '5432';
  const dbName = process.env.DB_NAME || 'postgres';
  const dbSsl = process.env.DB_SSL === 'true';
  
  if (!dbPassword) {
    console.error('ERROR: DB_PASSWORD environment variable must be set');
    console.error('For local development, create a .env file with this variable');
    process.exit(1);
  }
  
  const encodedPassword = encodeURIComponent(dbPassword);
  const connectionString = `postgresql://${dbUser}:${encodedPassword}@${dbHost}:${dbPort}/${dbName}?sslmode=${dbSsl ? 'require' : 'disable'}`;
  
  console.log('Testing database connection...');
  
  const client = new Client({
    connectionString: connectionString,
    ssl: dbSsl ? { rejectUnauthorized: false } : false
  });

  try {
    await client.connect();
    console.log('Connected successfully');
    const res = await client.query('SELECT NOW()');
    console.log('Query result:', res.rows[0]);
    await client.end();
  } catch (err) {
    console.error('Connection error:', err.message);
    process.exit(1);
  }
}

test();

File: test-db.py
import psycopg2
import os
from dotenv import load_dotenv

load_dotenv()

def test_conn():
    db_name = os.getenv('DB_NAME', 'postgres')
    db_user = os.getenv('DB_USER', 'postgres')
    db_password = os.getenv('DB_PASSWORD')
    db_host = os.getenv('DB_HOST', 'localhost')
    db_port = os.getenv('DB_PORT', '5432')
    db_ssl = os.getenv('DB_SSL', 'false').lower() == 'true'
    
    if not db_password:
        print("ERROR: DB_PASSWORD environment variable must be set")
        print("For local development, create a .env file with this variable")
        return
    
    try:
        conn = psycopg2.connect(
            dbname=db_name,
            user=db_user,
            password=db_password,
            host=db_host,
            port=db_port,
            sslmode="require" if db_ssl else "disable"
        )
        print("Connected successfully")
        cur = conn.cursor()
        cur.execute("SELECT NOW()")
        print("Query result:", cur.fetchone())
        cur.close()
        conn.close()
    except Exception as e:
        print("Connection error:", e)

if __name__ == "__main__":
    test_conn()
